tests/unit/config/test_base.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+
+from guidellm.config.base import (
+    Settings,
+    Environment,
+    LoggingSettings,
+    OpenAISettings,
+    ReportGenerationSettings,
+)
+
+
+@pytest.mark.unit
+def test_default_settings():
+    settings = Settings()
+    assert settings.env == Environment.PROD
+    assert settings.logging == LoggingSettings()
+    assert settings.openai == OpenAISettings()
+    assert (
+        settings.report_generation.source
+        == "https://guidellm.neuralmagic.com/local-report/index.html"
+    )
+
+
+@pytest.mark.unit
+def test_settings_from_env_variables(monkeypatch):
+    monkeypatch.setenv("GUIDELLM__env", "dev")
+    monkeypatch.setenv("GUIDELLM__logging__disabled", "true")
+    monkeypatch.setenv("GUIDELLM__OPENAI__API_KEY", "test_key")
+    monkeypatch.setenv("GUIDELLM__OPENAI__BASE_URL", "http://test.url")
+    monkeypatch.setenv("GUIDELLM__REPORT_GENERATION__SOURCE", "http://custom.url")
```

Comment:
https://docs.python.org/3/library/unittest.mock.html#patch-dict

```suggestion
@pytest.mark.unit
def test_settings_from_env_variables(mocker):
    mocker.path.dict(os.environ, {"GUIDELLM__ENV": "dev", ...})
```

Reply:
good catch, updated